Improve ProxmoxVE documentation (#43959)
home-assistant/home-assistant.github.io: source/_integrations/proxmoxve.markdown

## Proxmox permissions

This integration reads status and resource usage from Proxmox VE, and can perform actions using `button` entities.

{% important %}

> To keep things secure, create a dedicated Proxmox VE user for Home Assistant and only grant the permissions you need. In short, do not use the `root` account!

Reviewer: Maybe remove the exclamation mark? :)

{% endimportant %}

### Choose the right role

The minimum required permissions depend on what you want to do:

- **Monitor only** (sensors and binary sensors): The **Auditor** role (`PVEAuditor`) is typically sufficient.
- **Monitor and control** (button entities like start, stop, reboot): You will need a role that allows those actions, such as **User** (`PVEVMUser`) or another custom role that includes the required privileges.

If you are not sure which privileges you need for control in your Proxmox VE setup, start with `PVEVMUser`, confirm everything works, and then tighten the permissions by switching to a custom role. If you want full but responsible control start with `PVEVMAdmin`.

### Create a Home Assistant group

Before creating the user, you need to create a group for the user.
Privileges can be either applied to Groups or Roles.

1. Select **Datacenter**
2. Open **Permissions** and select **Groups**

> 3. Select the **Create** button above all the existing groups

Reviewer 2: ```suggestion
1. Select **Datacenter**.
2. Open **Permissions** and select **Groups**.
3. Select the **Create** button above all the existing groups.
```

> 4. Name the new group (e.g., `HomeAssistant`)

Reviewer 2: ```suggestion
4. Name the new group (for example, `HomeAssistant`).
```

> 5. Confirm **Create**

Reviewer 2: ```suggestion
5. Confirm **Create**.
```

### Add Group Permissions to all Assets

Assign the role you chose to the group at the root path (**/**) so it applies to all nodes, VMs, and containers.

1. Select **Datacenter**
2. Select **Permissions**
3. Open **Add** and select **Group Permission**

> 4. For **Path**, select **/**

Reviewer 2: ```suggestion
1. Select **Datacenter**.
2. Select **Permissions**.
3. Open **Add** and select **Group Permission**.
4. For **Path**, select **/**.
```

> 5. for **Group**,  select your Home Assistant group (`HomeAssistant`)

Reviewer 2: ```suggestion
5. For **Group**,  select your Home Assistant group (`HomeAssistant`).
```

6. For **Role**, select the role you want to use, like **PVEAuditor** (monitoring only) or **PVEVMUser** (monitoring plus basic actions).
7. Make sure **Propagate** is checked

> 8. Confirm **Create**

Reviewer 2: ```suggestion
7. Make sure **Propagate** is checked.
8. Confirm **Create**.
```

### Create a user for Home Assistant

Using the `pve` realm helps limit the account to API access, instead of Linux system authentication and remote (SSH) command line access.

{% important %}
If you plan to use the `pve` realm, make sure you select it during user creation and use the `@pve` suffix in Home Assistant (like `hass@pve`).
{% endimportant %}

1. Select **Datacenter**
2. Open **Permissions** and select **Users**
3. Select **Add**

> 4. Enter a username (e.g.,`hass`)

Reviewer 2: ```suggestion
1. Select **Datacenter**.
2. Open **Permissions** and select **Users**.
3. Select **Add**.
4. Enter a username (for example,`hass`).
```

> 5. Set the realm to **Proxmox VE authentication server** for `pve` (or **Linux PAM standard authentication** for `pam`)

Reviewer 2: ```suggestion
5. Set the realm to **Proxmox VE authentication server** for `pve` (or **Linux PAM standard authentication** for `pam`).
```

6. Enter a secure password (it can be complex as you will only need to copy/paste it into your Home Assistant configuration)

> 7. Select the group just created earlier (`HomeAssistant`) to grant access to Proxmox

Reviewer 2: ```suggestion
6. Enter a secure password (it can be complex as you will only need to copy/paste it into your Home Assistant configuration).
7. Select the group just created earlier (`HomeAssistant`) to grant access to Proxmox.
```

8. Ensure **Enabled** is checked and **Expire** is set to "never" (i.e. leave it blank)

> 9. Confirm **Add**

Reviewer 2: ```suggestion
8. Ensure **Enabled** is checked and **Expire** is set to "never" (for example, leave it blank).
9. Confirm **Add**.
```

- **Restart**: Restarts a VM/LXC.
- **Reboot**: Reboots a node.
- **Shutdown**: Shuts a node down.
- **Hibernate**: Puts a VM in hibernation; only available to VMs.
- **Reset**: Resets a VM; only available to VMs.

## Troubleshooting

### Buttons not working

> If you want to use the `button` entities to control power actions (start/stop/reboot and similar actions), the Proxmox VE user must have the required privileges for those actions (for example `VM.PowerMgmt` on the relevant path).  If monitoring works but button presses fail, assign a more permissive role (or create a custom role) and try again.

Reviewer 2: ```suggestion
If you want to use the `button` entities to control power actions (start/stop/reboot and similar actions), the Proxmox VE user must have the required privileges for those actions (for example, `VM.PowerMgmt` on the relevant path).  If monitoring works but button presses fail, assign a more permissive role (or create a custom role) and try again.
```
